caikit_computer_vision/data_model/image.py

```diff
@@ -0,0 +1,107 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data structures for representing images."""
+
+import alog
+from caikit.core import error_handler
+from caikit.core import DataObjectBase, dataobject
+import numpy as np
+from PIL import Image as PILImage
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+
+from caikit_computer_vision.data_model.backends import PilImageBackend
+
+log = alog.use_channel("DATAM")
+error = error_handler.get(log)
+
+@dataobject(package="caikit_data_model.caikit_computer_vision")
+class Image(DataObjectBase):
+    """Data model for an image object; this stores the image in the backend as a PIL image, with
+    convenience views to interact with the Image as other formats as needed.
+    """
+    image_data: Annotated[bytes, FieldNumber(1)]
+
+    def __init__(self, *args, **kwargs):
+        # For now, we always delegate to the Numpy image backend, because it's the only one.
+        # In the future, we may have more views etc, but for now, images as numpy arrays is
+        # the hub format.
+        #
+        # NOTE: Currently, the args / kwargs here should match up with the proto class, because
+        # .from_proto() requires it. In the future, it's probably good to have a generic utility
+        # in Watson Core to make sure that args / kwargs line up correctly and explode if they
+        # do not.
+        self._backend = PilImageBackend(*args, **kwargs)
```

Comment:
This would do the convert at initialization time, do we want to do it at conversion time (like in to_proto etc) ? Or is the idea is that we want to save the image internally in PIL format and then do the conversion to proto byte later on?